packages/utils/src/date-util/index.ts

```
@@ -0,0 +1,310 @@
+  const ndate = normalizeDate(date)
+  const nranges = ranges.map((range) => range.map(normalizeDate))
+
+  if (nranges.some((nrange) => ndate >= nrange[0] && ndate <= nrange[1])) {
+    return date
+  }
+
+  let minDate = nranges[0][0]
+  let maxDate = minDate
+
+  nranges.forEach((nrange) => {
+    let minTempDate = (minDate = new Date(Math.min(nrange[0], minDate)))
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)